src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,37 +1,64 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector((state) => state.currentTodo);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, []);
+
+  const handleClose = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading && <Loader />}
```

Comment:
you should display loader or modal content. not both of them:
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/102289654/cc4bb94f-0c93-49d6-af9c-22dfb6b23f36)